src/components/FeatureViews/Courses/ClassEnrollmentRow.js

```diff
@@ -113,6 +117,101 @@ const ClassEnrollmentRow = ({
         open: false,
     });
 
+    const [inviteStatus, setInviteStatus] = useState('Unsent');
+    const [
+        googleClassroomStatusMessage,
+        setGoogleClassroomStatusMessage,
+    ] = useState('Send Invite');
+    const { courses, google_courses, google_access_token } = useSelector(
+        ({ auth }) => auth
+    );
+
+    const getGoogleClassCode = (courses, courseID) => {
+        let googleClassCode;
+        if (courses) {
+            courses.forEach(function (course) {
+                if (courseID == course.id) {
+                    googleClassCode = course.googleClassCode;
+                }
+            });
+            return googleClassCode;
+        }
+    };
+
+    const getGoogleCourseID = (google_courses, courses, courseID) => {
+        let googleCourseID;
+        let googleClassCode = getGoogleClassCode(courses, courseID);
+        if (google_courses && googleClassCode) {
+            google_courses.forEach(function (course) {
+                if (course.enrollmentCode == googleClassCode) {
+                    googleCourseID = course.id;
+                }
+            });
+            return googleCourseID;
+        }
+    };
+
+    const handleInvite = async () => {
+        let googleCourseID = getGoogleCourseID(
+            google_courses,
+            courses,
+            courseID
+        );
+        if (
+            googleClassroomStatusMessage == 'Resend Invite' ||
+            inviteStatus == 'Unsent'
```

Comment:
why are you checking for these 2 statuses? also put these 2 statuses into an explanatory variable